### PR TITLE
test: Add all unit tests and gha workflow 

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

Add all unit tests for framework_models and helper functions

## ✔️ How to Test

make unittest

**How do I run the relevant unit/integration tests?**

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.19'
+
+      - run: go version
+
+      - name: Run smoke tests
+        run: make unittest
+        env:
+          LINODE_TOKEN: ${{ secrets.DX_LINODE_TOKEN }}
```
Comment (lgarber-akamai) [inline .github/workflows/unit-tests.yml:26]: Could this secret be dropped so it's not accidentally exposed in PR workflow runs?

**Review by lgarber-akamai — CHANGES_REQUESTED** [review]
All unit tests are passing locally and I confirmed that `make testacc` and `make unittest` only run their corresponding tests. Just one small change this should be good to go!

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (ykim-1) [reply .github/workflows/unit-tests.yml:26]: Yup I will remove this

**Comment by ykim-1** [comment]
Finally a successful run :) 

https://github.com/linode/terraform-provider-linode/actions/runs/5929290433/job/16076659841?pr=985

**Review by yec-akamai — APPROVED** [review]
Great work! 🎉

**Review by lgarber-akamai — APPROVED** [review]
Looks great!

**Pull request merged**